client/me/profile/index.jsx

```diff
@@ -120,7 +120,7 @@ class Profile extends Component {
 								checked={ this.props.getSetting( 'gravatar_profile_hidden' ) }
 								onChange={ this.toggleGravatarHidden }
 								label={ this.props.translate(
-									'{{spanLead}}Hide my Gravatar profile.{{/spanLead}} {{spanExtra}}This will prevent your {{profileLink}}Gravatar profile{{/profileLink}} and photo from appearing on any site. It may take some time for the changes to take effect. Gravatar profiles can be deleted at {{deleteLink}}Gravatar.com{{/deleteLink}}.{{/spanExtra}}',
+									'{{spanLead}}Hide my photo and Gravatar profile.{{/spanLead}} {{spanExtra}}This will prevent your {{profileLink}}Gravatar profile{{/profileLink}} and photo from appearing on any site. It may take some time for the changes to take effect. Gravatar profiles can be deleted at {{deleteLink}}Gravatar.com{{/deleteLink}}.{{/spanExtra}}',
```

Comment:
I sometimes like to be a little extra when deploying a string change like this where it's a block of text that's only slightly changed.

As soon as this deploys, because we don't have the translations for it yet, it's going to default to showing the English version for all non-English speaking users. Since the new message is only incrementally better than the old message, Spanish speakers for instance would be better off keeping the old message until translations are ready. It'd also be a big chunk of unreadable text appearing in a quite important part of the UI, so it might be a bit unnerving not knowing what it says!

```js
import i18n, { localize } from 'i18n-calypso';
import { englishLocales } from '@automattic/i18n-utils';

function MyComponent( { translate, locale } ) {
	const message = englishLocals.includes( locale ) && i18n.hasTranslation( 'The updated string' )
		? translate( 'The updated string' )
		: translate( 'The old string' );

	return <p>{ message }</p>;
}

// I'll use the localize HoC here to get access to `translate` and `locale` props
// since that's how this profile view works.
export default localize( MyComponent );
```

Whenever a translation becomes available in glotpress, it'll appear in the UI. Once all the mag16 languages are translated we can tidy up this code as a Janitorial task.

---

The simpler way of doing this is to simply not merge the PR until all translations are complete. This is doable with the `String Freeze` label. p4TIVU-5Jq-p2
But that might be a few weeks, and we presumably don't want to block this bug from being fixed for that long.

Reply:
> I sometimes like to be a little extra when deploying a string change like this where it's a block of text that's only slightly changed.

@p-jackson Great suggestion! I've applied an implementation of your example in 2600366850